### PR TITLE
Closure builder for inner managers of centralized.

### PR DESCRIPTION
**Pull request by rmalmain**
the closures are different for the main node and the secondary nodes for finer-grained configuration.

### CODE REVIEW

File: libafl/src/events/launcher.rs [hunk 1]
```diff
@@ -542,27 +545,93 @@ impl<CF, MF, MT, S, SP> Debug for CentralizedLauncher<'_, CF, MF, MT, S, SP> {
...
+        &mut self,
+        main_inner_mgr_builder: MIMF,
+        client_inner_mgr_builder: CIMF,
+    ) -> Result<(), Error>
+    where
+        CIMF: FnOnce(&Self, CoreId) -> Result<(Option<S>, CIM), Error>,
+        MIMF: FnOnce(&Self, CoreId) -> Result<(Option<S>, MIM), Error>,
+    {
+        let mut main_inner_mgr_builder = Some(main_inner_mgr_builder);
+        let mut client_inner_mgr_builder = Some(client_inner_mgr_builder);
```
Comment (tokatoka) [inline libafl/src/events/launcher.rs:633]: secondary_inner_mgr_builder?

i think main evaluator is `client` too,

Comment (rmalmain) [reply libafl/src/events/launcher.rs:633]: yes ur right its better

Comment (rmalmain) [reply libafl/src/events/launcher.rs:633]: i'll change other places where its not consistent with this notation

File: libafl/src/events/launcher.rs [hunk 2]
```diff
@@ -524,11 +527,11 @@ pub struct CentralizedLauncher<'a, CF, MF, MT, S, SP> {
     #[builder(default = LlmpShouldSaveState::OnRestart)]
     serialize_state: LlmpShouldSaveState,
     #[builder(setter(skip), default = PhantomData)]
-    phantom_data: PhantomData<(&'a S, &'a SP)>,
+    phantom_data: PhantomData<(&'a S, &'a SP, CIM, MIM)>,
 }
 
 #[cfg(all(unix, feature = "std", feature = "fork"))]
-impl<CF, MF, MT, S, SP> Debug for CentralizedLauncher<'_, CF, MF, MT, S, SP> {
+impl<CF, CIM, MF, MIM, MT, S, SP> Debug for CentralizedLauncher<'_, CF, CIM, MF, MIM, MT, S, SP> {
```
Comment (tokatoka) [inline libafl/src/events/launcher.rs:534]: imo cim should always be the same as mim. what do you think.
so we can delete mim and just keep cim

Comment (rmalmain) [reply libafl/src/events/launcher.rs]: ok

**Pull request merged**